templates/istio.yaml

```
@@ -0,0 +1,87 @@
+{{ if .Values.istio }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: istio-system
+---
+apiVersion: helm.cattle.io/v1
+kind: HelmChart
+metadata:
+  name: istio-cni
+  namespace: kube-system
+spec:
+  bootstrap: true
+  chart: https://istio-release.storage.googleapis.com/charts/cni-1.17.1.tgz
+  valuesContent: |
+    cni:
+      cniBinDir: /var/lib/rancher/k3s/data/current/bin
+      cniConfDir: /var/lib/rancher/k3s/agent/etc/cni/net.d
+---
+apiVersion: helm.cattle.io/v1
+kind: HelmChart
+metadata:
+  name: istio-base
+  namespace: istio-system
+spec:
+  bootstrap: true
+  chart: https://istio-release.storage.googleapis.com/charts/base-1.17.1.tgz
+---
+apiVersion: helm.cattle.io/v1
+kind: HelmChart
+metadata:
+  name: istiod
+  namespace: istio-system
+spec:
+  bootstrap: true
+  chart: https://acorn-io.github.io/hub-system-charts/istiod-1.17.1.tgz
+  valuesContent: |
+    pilot:
+      podAnnotations:
+        ad.datadoghq.com/discovery.checks: |
+          {
+            "istio": {
+              "init_config": {},
+              "instances": [
+                {
+                  "istiod_endpoint": "http://%%host%%:15014/metrics",
+                  "use_openmetrics": "true",
+                  "exclude_labels": [
+                    "source_version",
+                    "destination_version",
+                    "source_canonical_revision",
+                    "destination_canonical_revision",
+                    "source_principal",
+                    "destination_principal",
+                    "source_cluster",
+                    "destination_cluster",
+                    "source_canonical_service",
+                    "destination_canonical_service",
+                    "destination_workload_namespace",
+                    "request_protocol",
+                    "connection_security_policy",
+                    "destination_service"
+                  ]
+                }
+              ]
+            }
+          }
```

This annotation doesn't quite seem to be working for us yet - I'm in contact with Datadog support about that. For now, let's switch back to the older style. From testing things out, I know that this is at least sufficient to enable OpenMetrics v2, which is very important for the data to be accurate.

```suggestion
        ad.datadoghq.com/discovery.check_names: '["istio"]'
        ad.datadoghq.com/discovery.init_configs: '[{}]'
        ad.datadoghq.com/discovery.instances: |
          [
            {
              "istiod_endpoint": "http://%%host%%:15014/metrics",
              "use_openmetrics": "true"
            }
          ]
```

---

Done!